cmd/cluster/instance_migrate.go

```diff
@@ -582,3 +631,72 @@ func maybeNotifyForUnmanagedAnnotations(aviObj *unstructured.Unstructured) {
 
 	}
 }
+
+func updateAllStoredVersions(crds map[string][]string) error {
+	for crd := range crds {
+		if err := updateStoredVersionsTo(crd, crdsStoredVersionsAfterUpgrade); err != nil {
+			fmt.Printf("Error updating %s: %s. Will restore CRDs.", crd, err.Error())
+			return err
+		}
+	}
+	return nil
+}
+
+func restoreAllStoredVersions(crds map[string][]string) error {
+	fmt.Println("Restoring CRDs to the previous state...")
+	for crd, oldStoredVersions := range crds {
+		if err := updateStoredVersionsTo(crd, oldStoredVersions); err != nil {
+			return err
+		}
+	}
+	fmt.Println("Done. Please, check again the upgrade guide for Astarte Operator 1.1 before retrying.")
+	return nil
+}
+
+func updateStoredVersionsTo(crdName string, newStoredVersions []string) error {
+	// fetch the CRD
+	crd, err := getCRD(crdName)
+	if err != nil {
+		return err
+	}
+
+	// if it has the desired storedVersions already, do nothing
+	if checkStoredVersionsMatch(crd.Status.StoredVersions, newStoredVersions) {
+		return nil
+	}
+
+	// else, perform the update
+	if err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		crd, err = getCRD(crdName)
+		if err != nil {
+			return err
+		}
+
+		crd.Status.StoredVersions = newStoredVersions
+
+		_, err = kubernetesAPIExtensionsClient.ApiextensionsV1().CustomResourceDefinitions().UpdateStatus(context.Background(), crd, metav1.UpdateOptions{})
+		if err != nil {
+			return err
+		}
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	return nil
+}
```

Comment:
I think we can shrink this func. What about this?
```suggestion
func updateStoredVersionsTo(crdName string, newStoredVersions []string) error {
	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
		if crd, err := kubernetesAPIExtensionsClient.ApiextensionsV1().CustomResourceDefinitions().Get(context.Background(),
			crdName, metav1.GetOptions{}); err != nil {
			return err
		}

		if checkStoredVersionsMatch(crd.Status.StoredVersions, newStoredVersions) {
			return nil
		}

		crd.Status.StoredVersions = newStoredVersions

		if _, err = kubernetesAPIExtensionsClient.ApiextensionsV1().CustomResourceDefinitions().UpdateStatus(context.Background(), crd, metav1.UpdateOptions{}); e != nil {
			return err
		}

		return nil
	})
}
```